akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowDropWhileSpec.scala

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.stream.ActorOperationAttributes._
+import akka.stream.Supervision._
+import akka.stream.testkit.Utils._
+
+import akka.stream.ActorFlowMaterializer
+import akka.stream.ActorFlowMaterializerSettings
+import akka.stream.testkit._
+
+import scala.util.control.NoStackTrace
+
+class FlowDropWhileSpec extends AkkaSpec {
+
+  val settings = ActorFlowMaterializerSettings(system)
+
+  implicit val materializer = ActorFlowMaterializer(settings)
+
+  "A DropWhile" must {
+
+    "drop while predicate is true" in assertAllStagesStopped {
+      val subscriber = TestSubscriber.manualProbe[Int]()
+      Source(1 to 4).dropWhile(_ < 3).runWith(Sink.publisher).subscribe(subscriber)
```

Comment:
You could use `akka.stream.testkit.scaladsl.TestSink.probe` directly in the `.runWith` which gives a nice and fluent DSL for the following assertions.